### PR TITLE
Add \text_map_... functions

### PR DESCRIPTION
**Pull request by josephwright**
Currently missing at the implementation level are regional indicator and extended pictogram rules: to do later as the requirement becomes clearer.

I plan to add more tests based on https://www.unicode.org/Public/UCD/latest/ucd/auxiliary/GraphemeBreakTest.txt, but that will come in a separate commit as it needs a bit of separate work to automate fully.

### CODE REVIEW

**Comment by josephwright** [comment]
Note that at present this code recurses into brace groups. The thinking is that `\emph{Graphemes}` might be reasonable to handle, so we need recursion. Does leave open the question of what `\text_map_break:` should do there. Also I had to decided how to handle control sequences, and at the moment there is no special treatment for math more, awkward control sequence arguments, etc.

**Comment by josephwright** [comment]
(Really comes down to the fact that mapping to input like this really only makes sense if it's 'pure text' - once it's not, all hell breaks loose.)

**Comment by josephwright** [comment]
One might argue that like for case changing, rather than a mapping we could provide a 'transformation' approach, which therefore would avoid some of those questions. But it might be hard to define what it _does_ mean.

**Review by PhelypeOleinik — APPROVED** [review]

**Comment by josephwright** [comment]
@PhelypeOleinik  Thanks

**Comment by josephwright** [comment]
Other than the regional business, I think the test failures are now pretty low-priority: most seem to relate to text that a user would never give as they mix up chars from different alphabets.

File: l3kernel/testfiles/m3text006.lvt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+\START
+\AUTHOR{Joseph Wright}
+
+\ExplSyntaxOn
+
+\TEST { Grapheme~mapping }
+  {
+    \OMIT
+      \ior_open:Nn \g_tmpa_ior { GraphemeBreakTest.txt }
```
Comment (FrankMittelbach) [inline l3kernel/testfiles/m3text006.lvt:21]: That test file is a bit heavy without any documentation what it does or supposed to be tested. Could there be a few words up at the top  saying that?

File: l3kernel/testfiles/m3text006.tlg [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+TEST 1: Grapheme mapping
+============================================================
+TRUE
+TRUE
+TRUE
+TRUE
+TRUE
+TRUE
+TRUE
+TRUE
```
Comment (FrankMittelbach) [inline l3kernel/testfiles/m3text006.tlg:14]: if that ever says something else you have probably a hard time to figures out which TRUE is which :-)

**Review by FrankMittelbach — COMMENTED** [review]
only looked over the documentation parts

File: l3kernel/testfiles/m3text006.tlg (hunk 2 above)
Comment (josephwright) [reply l3kernel/testfiles/m3text006.tlg:14]: True, and but when it does that it's quite easy to add a `\TYPEOUT` and check

Comment (josephwright) [reply l3kernel/testfiles/m3text006.tlg:14]: Ah, I actually left that in the test, so a failure will print the text which is wrong.

Comment (FrankMittelbach) [reply l3kernel/testfiles/m3text006.tlg:14]: ok then :-)

**Pull request merged**